Python/graminit.c

```diff
@@ -2497,6 +2483,6 @@ static label labels[184] = {
 grammar _PyParser_Grammar = {
     92,
     dfas,
-    {184, labels},
+    {183, labels},
```

Comment:
Do you understand why the regenerated graminit.c is different? Is it just that pgen2 processes some nodes in a different order? Or is the algorithm actually different in some way? Hopefully the order is stable? (If it were to use sets, it might not be!)

Reply:
> Hopefully the order is stable? (If it were to use sets, it might not be!)

I can confirm that the order is stable, regardless of the Python version you use to run pgen2. The reason is because iteration over sets and regular dicts is stabilized by sorting the container first. I have removed the sorting as a test and make use of OrderedDicts everywhere and there are still differences with the pgen version of the files. 

> Do you understand why the regenerated graminit.c is different? Is it just that pgen2 processes some nodes in a different order? Or is the algorithm actually different in some way?

The labels have different order because the arcs in the dfas have different order (labels are generated by `make_label` from the arcs in the dfas generated by `self.parse()`). The arcs themselves also follow the order in which dfas are generated. Note that dfas themselves are not changed from one file to the other, so they are parsed in the same fashion. 

pgen seems to call  `addlabel` when an nfa is created and when an atom is compiled so that explains the different in the order of the labels and I think the majority of the file.

I did not put much effort in recreating the same order because the grammar files are equivalent and the test suite passes.